### PR TITLE
Updates Hl7.Fhir.Stu3 to version 1.2

### PR DESCRIPTION
**Pull request by brendankowitz**

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Tests.E2E/Rest/Search/DateSearchTestFixture.cs [hunk 1]
```diff
@@ -15,19 +15,33 @@ public class DateSearchTestFixture : HttpIntegrationTestFixture<Startup>
...
-                p => p.BirthDate = "1979-12-31",                // 1979-12-31T00:00:00.0000000 <-> 1979-12-31T23:59:59.9999999
-                p => p.BirthDate = "1980",                      // 1980-01-01T00:00:00.0000000 <-> 1980-12-31T23:59:59.9999999
-                p => p.BirthDate = "1980-05",                   // 1980-05-01T00:00:00.0000000 <-> 1980-05-31T23:59:59.9999999
-                p => p.BirthDate = "1980-05-11",                // 1980-05-11T00:00:00.0000000 <-> 1980-05-11T23:59:59.9999999
-                p => p.BirthDate = "1980-05-11T16:32",          // 1980-05-11T16:32:00.0000000 <-> 1980-05-11T16:32:59.9999999
-                p => p.BirthDate = "1980-05-11T16:32:15",       // 1980-05-11T16:32:15.0000000 <-> 1980-05-11T16:32:15.9999999
-                p => p.BirthDate = "1980-05-11T16:32:15.500",   // 1980-05-11T16:32:15.5000000 <-> 1980-05-11T16:32:15.5000000
-                p => p.BirthDate = "1981-01-01").Result;        // 1981-01-01T00:00:00.0000000 <-> 1981-12-31T23:59:59.9999999
+            Observations = FhirClient.CreateResourcesAsync<Observation>(
+                () => new Observation
```
Comment (jackliums) [inline test/Microsoft.Health.Fhir.Tests.E2E/Rest/Search/DateSearchTestFixture.cs]: did these become required?

File: test/Microsoft.Health.Fhir.Tests.E2E/Common/FhirClientExtensions.cs [hunk 2]
```diff
@@ -61,12 +61,18 @@ public static async Task<TResource> CreateResourcesAsync<TResource>(this FhirCli
 
         public static async Task<TResource[]> CreateResourcesAsync<TResource>(this FhirClient client, params Action<TResource>[] resourceCustomizer)
             where TResource : Resource, new()
+        {
+            return await CreateResourcesAsync(client, () => new TResource(), resourceCustomizer);
+        }
+
+        public static async Task<TResource[]> CreateResourcesAsync<TResource>(this FhirClient client, Func<TResource> setup, params Action<TResource>[] resourceCustomizer)
```
Comment (jackliums) [inline test/Microsoft.Health.Fhir.Tests.E2E/Common/FhirClientExtensions.cs]: Can you also update the rest of the text fixture to follow the same pattern? (e.g., NumberSearchTestFixture still uses the old way).

File: test/Microsoft.Health.Fhir.Tests.E2E/Rest/Search/DateSearchTestFixture.cs (hunk 1 above)
Comment (brendankowitz) [reply test/Microsoft.Health.Fhir.Tests.E2E/Rest/Search/DateSearchTestFixture.cs]: I changed the resource from Patient to Observation, which is why `status` is required. I'm also using a unique coding so this test doesn't conflict with existing/other observations in the db

File: test/Microsoft.Health.Fhir.Tests.E2E/Common/FhirClientExtensions.cs (hunk 2 above)
Comment (brendankowitz) [reply test/Microsoft.Health.Fhir.Tests.E2E/Common/FhirClientExtensions.cs]: This should now be aligned, I used the existing local function pattern

**Pull request merged**